supervisor/shared/tick.c

```diff
@@ -93,13 +91,11 @@ void supervisor_background_tasks(void *unused) {
 
     assert_heap_ok();
 
-    last_finished_tick = port_get_raw_ticks(NULL);
-
     port_finish_background_task();
 }
 
 bool supervisor_background_tasks_ok(void) {
-    return port_get_raw_ticks(NULL) - last_finished_tick < 1024;
+    return port_get_raw_ticks(NULL) - get_background_ticks() < 1024;
```

Comment:
I think we need to add a bit more smarts between this and background_callback. Lets have get_background_ticks be 0 when no tasks are pending. That changes here so background tasks are ok if it's 0 or the difference is < 1024.